arcgis-ios-sdk-samples/Edit data/Add features with contingent values/ContingentValuesTableViewController.swift

```diff
@@ -0,0 +1,303 @@
+    }
+    
+    /// Reset the cell states according to which values have already been selected.
+    func resetCellStates() {
+        if selectedStatus == nil {
+            protectionCell.textLabel?.isEnabled = false
+            protectionCell.isUserInteractionEnabled = false
+        } else {
+            protectionCell.textLabel?.isEnabled = true
+            protectionCell.isUserInteractionEnabled = true
+        }
+        if selectedProtection == nil {
+            bufferSizeCell.textLabel?.isEnabled = false
+            bufferSizeCell.isUserInteractionEnabled = false
+        } else {
+            bufferSizeCell.textLabel?.isEnabled = true
+            bufferSizeCell.isUserInteractionEnabled = true
+        }
+    }
+    
+    /// Toggles visibility of the buffer size scale picker.
+    func toggleBufferSizePickerVisibility() {
+        let bufferSizePicker = IndexPath(row: 3, section: 0)
+        tableView.performBatchUpdates({
+            if bufferSizePickerHidden {
+                tableView.insertRows(at: [bufferSizePicker], with: .fade)
+                bufferSizePickerHidden = false
+            } else {
+                tableView.deleteRows(at: [bufferSizePicker], with: .fade)
+                bufferSizePickerHidden = true
+            }
+        }, completion: nil)
+    }
+    
+    // MARK: UITableViewController
+    
+    /// Show the list of options according to the selected row.
+    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        let cell = tableView.cellForRow(at: indexPath)
+        switch cell {
+        case statusCell:
+            showStatusOptions()
+        case protectionCell:
+            showProtectionOptions()
+        case bufferSizeCell:
+            tableView.deselectRow(at: indexPath, animated: true)
+            showBufferSizeOptions()
+            toggleBufferSizePickerVisibility()
+        default:
+            return
+        }
+    }
+        
+    /// Return the number of rows depending on the picker view.
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        let numberOfRows = super.tableView(tableView, numberOfRowsInSection: section)
+        if bufferSizePickerHidden {
+            return numberOfRows - 1
+        } else {
+            return numberOfRows
+        }
+    }
+}
+
+// MARK: Pickerview
+
+extension ContingentValuesTableViewController: UIPickerViewDataSource {
+    func numberOfComponents(in pickerView: UIPickerView) -> Int {
+        return 1
+    }
+    
+    func pickerView(_ pickerView: UIPickerView, numberOfRowsInComponent component: Int) -> Int {
+        return bufferSizes.count
+    }
+}
+
+extension ContingentValuesTableViewController: UIPickerViewDelegate {
+    func pickerView(_ pickerView: UIPickerView, titleForRow row: Int, forComponent component: Int) -> String? {
+        let bufferSizeTitle = String(bufferSizes[row])
+        return bufferSizeTitle
```

Comment:
Ideally a number formatter would be used to format an integer as a string so that the number is localized. That might be a good thing to model (using a number formatter).

Reply:
I've updated this to use a number formatter but it is slightly more complicated since I have to convert the Int into an NSNumber first.